helm/chart/router/templates/_helpers.tpl

```diff
@@ -99,3 +99,20 @@ Usage:
 "/metrics"
 {{- end }}
 {{- end -}}
+
+{{/*
+This function takes the `extraLabels` map and templatizes the values.
+This allows to pass references from the values and interprates them as template.
+A use case for this usage is to add a custom label to the deployment referencing the version of the chart.
+For example:
+
+```
+extraLabels:
+  custom-version: {{ .Chart.AppVersion }}
+```
+*/}}
+{{- define "common.templatizeExtraLabels" -}}
+{{- range $key, $value := .Values.extraLabels }}
+{{ printf "%s: %s" $key (include  "common.tplvalues.render" ( dict "value" $value "context" $))}}
```

Comment:
Note that this does not quote the value, so the PR description is a bit inaccurate. Rather than generating

```
labels:
  env: "v1.2.3"
```

it generates

```
labels:
  env: v1.2.3
```

This is usually equivalent, but not if the value is something like `true` or `null` or (I think) a number.

What this means is that

```
router:
  extraLabels:
    foo: "true"
```

is converted to

```
labels:
  foo: true
```

which is not actually valid in k8s.